rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/azure/AzureBakeHandlerSpec.groovy

```diff
@@ -331,4 +331,34 @@ class AzureBakeHandlerSpec extends Specification implements TestDefaults{
     1 * imageNameFactoryMock.buildPackagesParameter(NUPKG_PACKAGE_TYPE, osPackages) >> NUPKG_PACKAGES_NAME
     1 * packerCommandFactoryMock.buildPackerCommand("", parameterMap, null, "$configDir/azure-windows-2012-r2.json")
   }
+
+  void 'Create proper azure_image_name'() {
+    setup:
+    def azureBakeHandler = new AzureBakeHandler(azureBakeryDefaults: azureBakeryDefaults)
+
+    when:
+    def bakeRequest = new BakeRequest(user: "someuser@gmail.com",
+      package_name: NUPKG_PACKAGES_NAME,
+      base_os: "windows-2012-r2",
+      cloud_provider_type: BakeRequest.CloudProviderType.azure,
+      build_number: buildNumber,
+      base_name: baseName)
+    def azureVirtualizationSettings = azureBakeHandler.findVirtualizationSettings(REGION, bakeRequest)
+    def parameterMap = azureBakeHandler.buildParameterMap(REGION, azureVirtualizationSettings, imageName, bakeRequest, "SomeVersion")
+
+    then:
+    parameterMap.azure_image_name == expectedAzureImageName
+
+    where:
+    buildNumber   || baseName    || imageName                                           || expectedAzureImageName
```

Comment:
This should look like:
`buildNumber   | baseName    | imageName                                           || expectedAzureImageName`
for each line.

It is just a convention to separate the givens from the expected. I only learned this recently.